### PR TITLE
[MLIR][Affine]Fixed crash with invalid loop bound (Issue #64287)

### PR DESCRIPTION
**Pull request by SwapnilGhanshyala**

### CODE REVIEW

File: mlir/lib/Dialect/Affine/Analysis/AffineStructures.cpp [hunk 1]
```diff
@@ -222,8 +221,17 @@ LogicalResult FlatAffineValueConstraints::addBound(BoundType type, unsigned pos,
   fullyComposeAffineMapAndOperands(&map, &operands);
   map = simplifyAffineMap(map);
   canonicalizeMapAndOperands(&map, &operands);
-  for (auto operand : operands)
+  for (auto operand : operands) {
+    if (!((isTopLevelValue(operand)) || (isAffineInductionVar(operand)))) {
```
Comment (bondhugula) [inline mlir/lib/Dialect/Affine/Analysis/AffineStructures.cpp]: Unnecessary parentheses here. Use `!.... && !....`.

File: mlir/lib/Dialect/Affine/Analysis/AffineStructures.cpp [hunk 2]
```diff
@@ -222,8 +221,17 @@ LogicalResult FlatAffineValueConstraints::addBound(BoundType type, unsigned pos,
...
+      emitError(
+          operand.getLoc(),
+          " Above Val is neither defined as a Top-Level Value nor is an "
+          "AffineInductionVar. Non-terminal symbol / loop IV expected. Pass "
+          "Aborted!");
```
Comment (bondhugula) [inline mlir/lib/Dialect/Affine/Analysis/AffineStructures.cpp]: Remove unnecessary capitatlization. All emit messages should start in lower case.

"value is neither a top-level value nor an affine IV; non-termnial symbol/loop IV expected; pass aborted"

File: mlir/lib/Dialect/Affine/Analysis/AffineStructures.cpp [hunk 3]
```diff
@@ -222,8 +221,17 @@ LogicalResult FlatAffineValueConstraints::addBound(BoundType type, unsigned pos,
...
+      return failure();
```
Comment (bondhugula) [inline mlir/lib/Dialect/Affine/Analysis/AffineStructures.cpp]: You can simply do `return emitError(...)` in one line.

File: mlir/lib/Dialect/Affine/Analysis/AffineStructures.cpp (hunk 2 above)
Comment (bondhugula) [inline mlir/lib/Dialect/Affine/Analysis/AffineStructures.cpp]: Putting an emitError is not the right fix - this method is simply meant to return failure. It's the caller that has to deal with it appropriately.

**Review by bondhugula — CHANGES_REQUESTED** [review]
This isn't the right fix.

Also, please update the commit summary to include a link to the issue being addressed by this if that's the case.

**Comment by SwapnilGhanshyala** [comment]
Analyzing the proposed fix. Will create new PR when done

**Pull request closed without merging**